### PR TITLE
[ANCHOR-214] [ANCHOR-215] Add docker compose to junit and cleanup docker-compose files

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

- Add `docker-compose-rule` to AnchorPlatformIntegrationTest.
- Remove singleton pattern from starting the servers to avoid the interaction between tests.
- clean up un-used docker-compose files, assets, stellar.toml resources.

### Why

To help development efforts and prepare for end-2-end Kotlin test scripts.

### Known limitations

N/A

### CODE REVIEW

File: kotlin-reference-server/src/main/kotlin/org/stellar/reference/RefenreceServerStart.kt [hunk 1]
```diff
@@ -2,4 +2,14 @@ package org.stellar.reference
 
 // Used in ServiceRunner
 @JvmOverloads // Annotation required to call from Java with optional argument
-fun start(waitServer: Boolean = false) = main(arrayOf(waitServer.toString()))
+fun start(waitServer: Boolean = false) = startServer(waitServer)
+fun stop() = stopServer()
+
+class KotlinTest {
```
Comment (Ifropc) [inline kotlin-reference-server/src/main/kotlin/org/stellar/reference/RefenreceServerStart.kt:8]: Should be removed?

**Review by reecexlm — APPROVED** [review]
LGTM

File: kotlin-reference-server/src/main/kotlin/org/stellar/reference/RefenreceServerStart.kt (hunk 1 above)
Comment (lijamie98) [reply kotlin-reference-server/src/main/kotlin/org/stellar/reference/RefenreceServerStart.kt:8]: Yes, removed.

File: kotlin-reference-server/src/main/kotlin/org/stellar/reference/ReferenceServer.kt [hunk 2]
```diff
@@ -42,16 +47,21 @@ fun main(args: Array<String>) {
...
+              anyHost()
+              allowHeader(HttpHeaders.Authorization)
+              allowHeader(HttpHeaders.ContentType)
+            }
+          }
+          .start(wait)
+}
+
+fun stopServer() {
+  (referenceKotlinSever)?.stop(1000, 1000)
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply kotlin-reference-server/src/main/kotlin/org/stellar/reference/ReferenceServer.kt]: Added.

**Pull request merged**